internal/fs/inode/dir.go

```diff
@@ -771,15 +771,19 @@ func (d *dirInode) CreateChildFile(ctx context.Context, name string) (*Core, err
 	}, nil
 }
 
-func (d *dirInode) CreateLocalChildFile(name string) (*Core, error) {
+func (d *dirInode) CreateLocalChildFile(name string, updateInodeFn func(core Core) error) error {
 	fullName := NewFileName(d.Name(), name)
-
-	return &Core{
+	core := Core{
 		Bucket:    d.Bucket(),
 		FullName:  fullName,
 		MinObject: nil,
 		Local:     true,
-	}, nil
+	}
+	if err := updateInodeFn(core); err != nil {
```

Comment:
Also can you add some details in description why all these changes are required.

Reply:
Refactored it to make the code more obvious.